test/unit/test_exports/test_exports.py

```diff
@@ -139,3 +140,39 @@ def test_instanciate_with_no_elements():
     """
     # define exports
     festim.Exports()
+
+
+def test_show_units_true_with_retention():
+    """Test to catch the bug in issue #765"""
+    my_model = festim.Simulation()
+    my_model.mesh = festim.MeshFromVertices(np.linspace(0, 1, num=200))
+
+    my_model.materials = festim.Material(id=1, D_0=1, E_D=0)
+
+    my_model.traps = festim.Trap(
+        k_0=1,
+        E_k=0,
+        p_0=1,
+        E_p=0,
+        density=1.3e-3,
+        materials=my_model.materials[0],
+    )
+
+    my_model.T = festim.Temperature(500)
+
+    my_model.settings = festim.Settings(
+        absolute_tolerance=1e10, relative_tolerance=1e-09, transient=False
+    )
+
+    my_model.exports = [
+        festim.DerivedQuantities(
+            [
+                festim.TotalVolume("retention", volume=1),
+                festim.TotalSurface("retention", surface=1),
+            ],
+            show_units=True,
+        )
+    ]
+
+    my_model.initialise()
+    my_model.run()
```

Comment:
This could potentially be simplified to not have a whole system test:
```suggestion
def test_show_units_true_with_retention():
    """Test to catch the bug in issue #765"""
     test_mesh = f.UnitIntervalMesh(10)

    my_exports = [
        F.TotalVolume(volume=3, field="retention"),
        F.TotalSurface(surface=2, field="retention"),
    ]
    for export in my_exports:
        export.dx = f.Measure("dx", domain=test_mesh)
        export.ds = f.Measure("ds", domain=test_mesh)
        export.show_units = True

    expected_titles = [
        "Total retention volume 3 (H m-2)",
        "Total retention surface 2 (H m-2)",
    ]

    for export, expected_title in zip(my_exports, expected_titles):
        assert export.title == expected_title
```

Reply:
This does not catch the bug described in the original issue. Running this on main produces:

```

    @property
    def export_unit(self):
        # obtain domain dimension
>       dim = self.function.function_space().mesh().topology().dim()
E       AttributeError: 'NoneType' object has no attribute 'function_space'
```

Which is not the bug. Moreover, one could fabricate the retention sum function but then this completely circumvent the integration testing.